File: java_solutions/830.positions-of-large-groups.java
/*
 * @lc app=leetcode id=830 lang=java
 *
 * [830] Positions of Large Groups
 */

// semicolons : 1
import java.util.regex.*;

class Solution {
    public List<List<Integer>> largeGroupPositions(String s) {
        return Pattern.compile("(.)\\1*").matcher(s).results()
            .filter(m -> m.end() - m.start() >= 3)
            .map(m -> Arrays.asList(m.start(), m.end() - 1))
            .toList();
    }
}

File: kotlin_solutions/830.positions-of-large-groups.kt
/*
 * @lc app=leetcode id=830 lang=kotlin
 *
 * [830] Positions of Large Groups
 */

class Solution { fun largeGroupPositions(s: String): List<List<Int>> = (s + " ").foldIndexed(mutableListOf<List<Int>>() to (s[0] to 0)) { i, (res, cur), c -> if (c == cur.first) res to (cur.first to cur.second) else (if (i - cur.second >= 3) res.apply { add(listOf(cur.second, i - 1)) } else res) to (c to i) }.first }
